libmscore/scale.cpp

```diff
@@ -0,0 +1,265 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2011-2012 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENSE.GPL
+//=============================================================================
+
+#include "scale.h"
+#include "mscore.h"
+#include "note.h"
+#include <set>
+
+using namespace std;
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   getNextLine
+//---------------------------------------------------------
+
+QString Scale::getNextLine(QTextStream& in)
+      {
+      while (!in.atEnd()) {
+            QString line = in.readLine();
+            if (line[0] == '!')
+                  continue;
+            return line;
+            }
+      return QString();
+      }
+
+//---------------------------------------------------------
+//   loadScale
+//---------------------------------------------------------
+
+bool Scale::loadScale(const QString& fn)
+      {
+      static set<int> supportedNbs = { 8, 12, 21, 35 };
+
```

Comment:
It should be 7. Thanks for finding the bug. It was working for me because I was adding 1 to the number of notes.